include/exiv2/asfvideo.hpp

```diff
@@ -47,7 +46,7 @@ namespace Exiv2 {
     /*!
       @brief Class to access ASF video files.
      */
-    class EXIV2_DEPRECATED EXIV2API AsfVideo:public Image
+    class EXIV2LIB_DEPRECATED_EXPORT AsfVideo:public Image
```

Comment:
Where does this macro come from?

Reply:
exiv2lib_export.h

Reply:
Ah, thanks. But then, how does this help with your problem? The macro `EXIV2LIB_DEPRECATED_EXPORT` is defined as `EXIV2API EXIV2_DEPRECATED` (on my machine), so unless you want to forcefully redefine it, won't you run into the same problem with the class with the `__attribute__` and `[]` syntax?

Reply:
@D4N  Luckly both `EXIV2API`  and `EXIV2_DEPRECATED` always expand into `__attribute__` form, so there won't be a mix of `__attribute__` and `[[]]`